rest/CONTRIBUTING.md

```diff
@@ -46,6 +46,32 @@ In this manner, we can ensure that the dependency chain for the oshinko-rest
 application will be frozen around the sources that are contained in the
 vendor directory.
 
+### Godeps
+
+To improve the provenance of the dependencies that are carried within the
+`vendor` directory, the `Godeps/Godeps.json` has been added to record the
+individual packages and commit hashes for each dependency. This file is
+usually generated by the `godep` tool, but as this project contains
+dependencies on the OpenShift Origin project(which carries its own modified
+dependencies), the `GoDeps.json` file has been created in a custom manner.
+
+The dependencies for this project have been seeded by first copying the
+`vendor` directory from the upstream OpenShift Origin project
+(https://github.com/openshift/origin), the `Godeps` directory is also copied.
+After the initial copy, the Origin project itself is copied into the vendor
+and its `vendor`, `Godeps` and `.git` directories are removed. Finally, as
+the rest server code contains a few extra dependencies, those are added
+manually by copying them from github into the appropriate locations in the
+vendor directory and their entries added to the `Godeps.json` file.
+
+*Dependency creation steps*
+1. copy `vendor` directory from `openshift/origin` to root of oshinko-cli
+1. copy `Godeps` directory from `openshift/origin` to root of oshinko-cli
+1. copy `openshift/origin` to `./vendor/github.com/openshift/`
+1. remove `vendor`, `Godeps` and `.git` directories from the copied
+   `./vendor/github.com/openshift/origin/` directory
+1. add rest dependencies to `vendor` directory and `Godeps.json` file
```

Comment:
should we list what the extra REST dependencies are at the moment, or at least how to determine what they are?